Theme weight

aydun opened the pull request:
Change the css file loading order so that the theme css overrides the other core css files.  The theme's css is still overridden by the Custom CSS setting allowing for local tweaks.

Depending on the progress of https://lab.civicrm.org/dev/core/-/issues/2460 this might become redundant.

aydun committed:
Increase theme css weight to override core css

aydun committed:
Simplify

vingle commented:
Thanks for this. Is the intended outcome that the theme's civicrm.css would load after, for e.g., civi's dashboard.css? That doesn't seem to happen with a test on Joomla. Or is it more about the interaction with the CMS admin theme css?

aydun committed:
Angular modules get added with weights 200+ so bump theme to 290
This means custom CSS also needs to be bumped above 290.

Until https://github.com/civicrm/civicrm-core/pull/20278 is merged,
there is no easy way to identify which snippet is the custom CSS one so
we search for 'custom.css' in the name.

aydun commented:
Yes, the theme's civicrm.css should load after dashboard.css.  CSS files added by Angular modules start at a weight of 200 (and dashboard.css is added by Angular) so the update bumps the weights further.

vingle commented:
That works. Nice fix, thanks. I think changing the order with the encapsulated css files and admin theme css might have some consequences, so want to test on each cms and maybe remove some `!important` declarations. Rather than have this on a new branch for that, I've just released v0.12, so will merge and test for a bit.
